src/circuit/circuit_frame.rs

```diff
@@ -3075,6 +3439,108 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
     Ok((result_expr, result_env, result_cont, make_thunk_num))
 }
 
+fn hide<F: LurkField, CS: ConstraintSystem<F>>(
+    mut cs: CS,
+    secret: F,
+    maybe_payload: &AllocatedPtr<F>,
+    store: &Store<F>,
+) -> Result<AllocatedPtr<F>, SynthesisError> {
+    let hide_ptr = if let Some(ptr) = maybe_payload.ptr(store).as_ref() {
+        match store.hidden(secret, *ptr) {
+            Some(c) => c,
+            None => store.get_nil(),
+        }
+    } else {
+        store.get_nil()
+    };
+    AllocatedPtr::alloc_ptr(&mut cs.namespace(|| "hide"), store, || Ok(&hide_ptr))
+}
+
+fn open<F: LurkField, CS: ConstraintSystem<F>>(
+    mut cs: CS,
+    maybe_commit: &AllocatedPtr<F>,
+    store: &Store<F>,
+) -> Result<AllocatedPtr<F>, SynthesisError> {
+    let hash = match maybe_commit.hash().get_value() {
+        Some(h) => h,
+        None => F::zero(),
```

Comment:
See comment above in definition of `hide`.

Reply:
Actually, this is a different situation, although maybe requiring the same treatment. For `hash` to be `None`, I think `maybe_commit` needs to 'contain' a `None` value because a blank circuit is being synthesized. This is, I think (once again), the horrible ambiguity of `bellman`, where `Option`s are used in this way. I think we still need a dummy value though.

Beyond the simple problem with clariy, I'm concerned that if we don't clearly mark dummy vs significant values we will start either:
- cargo-culting the significance of our chosen arbitrary values
- being inconsistent in ways that make it hard to understand that they *are* dummy values
- accidentally writing code which is only incidentally correct because it relies on some property of the dummy